File: WorldFeed/Client/src/app/core/Core/Iterators/Models/MultiLevelCursorIteratorPage.ts
import {CursorPageResult} from "./CursorPageResult";
import {IMultiLevelCursorIteratorPage} from "../../../Public/Iterators/IMultiLevelCursorIteratorPage";

export class MultiLevelCursorIteratorPage<TParent, TItem, TCursor> extends CursorPageResult<TItem, TCursor> implements IMultiLevelCursorIteratorPage<TParent, TItem, TCursor> {
  public associatedParentItems: TParent[];
}

File: Sharebook/Server/src/Sharebook.Common/Iterators/Models/MultiLevelCursorIteratorPage.cs
namespace Sharebook.Common.Iterators.Models
{
    using Sharebook.Common.Iterators.Models;
    using Sharebook.Common.Public.Iterators;

    public class MultiLevelCursorIteratorPage<TParent, TItem, TCursor> : CursorPageResult<TItem, TCursor>, IMultiLevelCursorIteratorPage<TParent, TItem, TCursor>
    {
        public TParent[] AssociatedParentItems { get; set; }
    }
}
